File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/movie_list/MovieListActivity.java
package com.example.lifecyclesandbox.movie_list;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.lifecyclesandbox.R;

import java.util.ArrayList;

public class MovieListActivity extends AppCompatActivity {
    private static final String TAG = "Activity_2";
    private ArrayList<Movie> movieList = arrayMovieList();
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_2);
        recyclerView = findViewById(R.id.movieListRecyclerView);
        Log.d(TAG, "onCreate(): " + MovieListActivity.class.getCanonicalName());

        setAdapter();
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart(): " + MovieListActivity.class.getCanonicalName());
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume(): " + MovieListActivity.class.getCanonicalName());
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG, "onRestart(): " + MovieListActivity.class.getCanonicalName());
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause(): " + MovieListActivity.class.getCanonicalName());
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop(): " + MovieListActivity.class.getCanonicalName());
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy(): " + MovieListActivity.class.getCanonicalName());
    }

    private ArrayList<Movie> arrayMovieList() {
        // TODO: Fix
        ArrayList<Movie> movieList = new ArrayList<>();

        movieList.add(new Movie("Star Wars"));
        movieList.add(new Movie("El Padrino"));
        movieList.add(new Movie("The Matrix"));

        return movieList;
    }

    private Movie fetchMovieData() {
        // TODO: fetch data from the API
        return null;
    }

    private void setAdapter() {
        MoviesAdapter adapter = new MoviesAdapter(movieList);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(adapter);
    }
}

File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/movie_list/MovieListActivity.kt
package com.example.lifecyclesandbox.movie_list

import android.os.Bundle
import android.util.Log
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import com.example.lifecyclesandbox.R
import com.example.lifecyclesandbox.movie_list.shared.MovieListViewModel
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

class MovieListActivity : AppCompatActivity() {

    private lateinit var movieListCollectionAdapter: MovieListCollectionAdapter
    private lateinit var viewPager: ViewPager2
    private val movieListViewModel : MovieListViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.hide()
        setContentView(R.layout.movie_list_activity)

        // TODO: Find out why pages are behaving like that (Favorite loads up info. loaded on Popular fragment, but does not update whenever info displayed on Popular changes).

        // Tab titles
        val tabTitles = arrayOf("popular", "favorite")

        // ViewPager
        movieListCollectionAdapter = MovieListCollectionAdapter(this, tabTitles.size)
        viewPager = this.findViewById(R.id.pager)
        viewPager.adapter = movieListCollectionAdapter

        val tabLayout = findViewById<TabLayout>(R.id.tab_layout)
        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
            tab.text = tabTitles[position]
        }.attach()
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart(): " + MovieListActivity::class.java.canonicalName)
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume(): " + MovieListActivity::class.java.canonicalName)
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "onRestart(): " + MovieListActivity::class.java.canonicalName)
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause(): " + MovieListActivity::class.java.canonicalName)
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop(): " + MovieListActivity::class.java.canonicalName)
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy(): " + MovieListActivity::class.java.canonicalName)
    }

    companion object {
        private const val TAG = "MovieListActivity"
    }
}